assignments/prototypes.js

```diff
@@ -16,13 +18,32 @@
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
 
+function GameObject(createdAt, name, dimensions) {
+  this.createdAt = createdAt;
+  this.name = name;
+  this.dimensions = dimensions;      
+}
+GameObject.prototype.destroy = function() {
+  return `${this.name} was removed from the game.`;
+}
+
 /*
   === CharacterStats ===
   * healthPoints
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
 
+function CharacterStats(healthPoints) {
+  GameObject.call(this);
+  this.healthPoints = healthPoints;  
+}
+
+CharacterStats.prototype = Object.create(GameObject.prototype);
+CharacterStats.prototype.takeDamage = function () {
+  return `${mage.name} took damage.`;
```

Comment:
```suggestion
  return `${this.name} took damage.`;
```

Reply:
now it will look for the value of name property of the object that the function is being called on. Reusability